haproxy/datadog_checks/haproxy/haproxy.py

```diff
@@ -202,6 +214,21 @@ def _collect_version_from_http(self, url):
             version = re.search(r"HAProxy version ([^,]+)", raw_version).group(1)
             self.log.debug("HAProxy version is %s", version)
             self.set_metadata('version', version)
+        if raw_uptime == "":
+            self.log.debug("unable to find HAProxy uptime")
+        else:
+            # It is not documented whether this output format is under any
+            # compatibility guarantee, but it hasn't yet changed since it was
+            # introduced
+            upinfo = re.search(r"(\d+)d (\d+)h(\d+)m(\d+)s", raw_uptime)
+            uptime = (
+                int(upinfo.group(1)) * 86400
+                + int(upinfo.group(2)) * 3600
+                + int(upinfo.group(3)) * 60
+                + int(upinfo.group(4))
+            )
```

Comment:
Can we put this in a function and document the calculation?

Reply:
Tell me if the new version is clearer: I named the groups (like `int(matched_uptime.group('days')) * 86400`) so that it is clear what each part of the addition means